tests/browser/async_stack_overflow.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <stdio.h>
+#include <emscripten.h>
+
+int main() {
+  EM_ASM({
+    window.onerror = function(e) {
+      var message = e.toString();
+      var success = message.indexOf("unreachable") >= 0 || // firefox
+                    message.indexOf("Script error.") >= 0; // chrome
+      if (success && !Module.reported) {
+        Module.reported = true;
+        console.log("reporting success");
+        // manually REPORT_RESULT; we shouldn't call back into native code at this point
+        var xhr = new XMLHttpRequest();
+        xhr.open("GET", "http://localhost:8888/report_result?0");
+        xhr.onload = xhr.onerror = function() {
+          window.close();
+        };
+        xhr.send();
+      }
+    };
+  });
+
+  volatile int x = 0;
+  volatile int y = 0;
+  volatile int z = 0;
```

Comment:
Can we use something other than volatile here?    `volatile` is pretty bad code smell I'd like to avoid if possible.

Reply:
I'll remove the volatile, it's not even optimized anyhow, so locals will remain.

Reply:
ok.  makes sense.